src/exchange/providers/ollama.py

```diff
@@ -0,0 +1,95 @@
+import os
+from typing import Any, Dict, List, Tuple, Type
+
+import httpx
+
+from exchange.message import Message
+from exchange.providers.base import Provider, Usage
+from exchange.providers.utils import (
+    messages_to_openai_spec,
+    openai_response_to_message,
+    openai_single_message_context_length_exceeded,
+    raise_for_status,
+    tools_to_openai_spec,
+)
+from exchange.tool import Tool
+
+OLLAMA_HOST = "http://localhost:11434/"
+
+
+
+#
+# NOTE: this is experimental, best used with 70B model or larger if you can
```

Comment:
Since experimental, we should flag it as such and raise a warning when instantiated with something like "This is an experimental provider and support may be dropped in the future" or something like that.

Reply:
yep please take a look again